docs/design/fabric-v2+/trusted_ledger.md

```diff
@@ -0,0 +1,162 @@
+- TLCC must be synronized with peer's ledger state
+- Secure (authenticated) channel between ECC and TLCC to query metadata;
+- ECC tx must perform read/write on a stable view of the ledger state
+    - (snapshot during a single transaction invocation)
+- TLCC must detect read/write inconsistency
+- TLCC must be able to perform identity validation (orderer, msp, etc...)
+- TLCC must provide channel metadata (chaincode definition)
+- Maintaining FPC chaincode state only is sufficient
+- Validate "normal" Enclave Registry (ERCC) transactions; however, ERCC comes with "hard-coded" endorsement policy
+
+
+## Fabric high-level validation 
+
+TODO pseudo code here
+
+Config validation: [source](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/common/configtx/validator.go#L163)
+
+- check config sequence number increased by 1
+- check is authorized Update [source](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/common/configtx/update.go#L115)
+    - verify ReadSet [source](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/common/configtx/update.go#L18)
+    - verify DeltaSet [source](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/common/configtx/update.go#L68)
+        - for each item validate policy [source](https://github.com/hyperledger/fabric/blob/f27912f2f419c3b35d2c1df120f19585815eceb0/common/policies/policy.go#L133)
+
+Lifecycle validation:
+
+
+Default validation:
+
+- Validating identities that signed the transaction
+    - read/write check?
+- Verifying the signatures of the endorsers on the transaction
+    - can endorse?
+- Ensuring the transaction satisfies the endorsement policies of the namespaces of the corresponding chaincodes.
+
+
+## Approach
+
+Restrict Fabric functionality:
+
+- keep Fabric validation changes in sync with TLCC
+- Reduces validation logic to a minimum
+    - minimizes catch-up 
+- Restrict the notion of endorsement policy; 
+- Simplify chaincode versioning (TODO)
+    - No chaincode updates for now
+- Support for default MSP only
+
+
+Any transactions/blocks with unsupported features are ignored/aborted
+
+Defined process to develop TLCC:
+- keep redundant validation code (in stock peer and TLCC) in sync
+- keep code relation traceable and changes trackable
+- consistent naming and code structure as much as possible
+- bonus: 
+    - automated code-changes notification; notifies whenever relevant Fabric code changes and might FPC must be updates.
+
+### Non-features
+
+- No state-based endorsement
+- No custom MSP (no idemix)
+- No support for custom endorsing/validation plugins for non-FPC and FPC chaincodes
+- Authentication and decorators
+- TODO check for more non-features in RFC
+
+## Design
+
+### Chaincode execution support
+    - secure channel
+    - validate proposal
+        - check org is "write"
+        - replay protection
```

Comment:
the view consistency/snapshot probably also belongs here?  

In that context, would be interesting to know how fabric currently does it: do they serialize update and endorsement or do they maintain different ledger states/views concurrently? the former is of course the easiest to implement (though requires an explicit notification by ecc that a tx is finished) but also means that a slow tx can hold of others (not that this is probably immediately the biggest concern for us :-))

(this in turn might ask for adding eventually the endorsement equivalent of the "Fabric high-level validation" section above?)